internal/test/framework/windowsvm.go

```diff
@@ -51,57 +52,114 @@ type TestWindowsVM interface {
 	types.WindowsVM
 }
 
-// newWindowsVM creates and sets up a Windows VM in the cloud and returns the WindowsVM interface that can be used to
-// interact with the VM. If credentials are passed then it is assumed that VM already exists in the cloud and those
-// credentials will be used to interact with the VM. If no error is returned then it is guaranteed that the VM was
-// created and can be interacted with. If skipSetup is true, then configuration steps are skipped.
-func newWindowsVM(imageID, instanceType string, credentials *types.Credentials, skipSetup bool) (TestWindowsVM, error) {
-	w := &testWindowsVM{}
-	var err error
-
-	cloudProvider, err = cloudprovider.CloudProviderFactory(kubeconfig, awsCredentials, "default", artifactDir,
-		imageID, instanceType, sshKey, privateKeyPath)
+//createMachineSet() gets the generated MachineSet configuration from cloudprovider package and creates a MachineSet
+func (f *TestFramework) createMachineSet() error {
+	cloudProvider, err := providers.NewCloudProvider(sshKey)
+	if err != nil {
+		return fmt.Errorf("error instantiating cloud provider %v", err)
+	}
+	machineSet, err := cloudProvider.GenerateMachineSet(true, 1)
```

Comment:
why not initialize f.machineSet here?

Reply:
in case we get an error machineSet will be set to nil, so assigning it after check. IMO, just noticed, i should be assigning it after creation is successful.